internal/pkg/runtime/engine/config/oci/generate/generate.go

```diff
@@ -110,6 +110,13 @@ func (g *Generator) AddProcessEnv(env, value string) {
 	g.Config.Process.Env = append(g.Config.Process.Env, kenv+value)
 }
 
+// SetProcessEnvWithPrefixes adds or replaces a container process environment variable for each prefix provided
```

Comment:
How about adding one more function named something like `SetProcessPrefixedEnv` which calls `SetProcessEnvWithPrefixes(env.ApptainerPrefixes, ...` to simply those repeated calls.

Reply:
@DrDaveD time to leave `good enough` alone ... introducing the change you suggest creates a circular reference between modules which takes even more refactoring to resolve